test/tic_tac_toe/ai_test.clj

```diff
@@ -73,54 +68,61 @@
 
 (deftest possible-board-state
   (testing "return a posible board state based on input"
-   (let [board  [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "_"]]]
-    (is (= [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "x"]] (possible-board 9 (@place-holder :player-marker) board))))))
+   (let [board  [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "_"]]
+          marker "x"]
+    (is (= [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "x"]] (possible-board 9 marker board))))))
+
+(deftest possible-game-state
+  (testing "returns board states based on input"
+   (let [ game { :board [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "_"]] :ai-marker "o" :player-marker "x"}]
+     (is (= '({:board [["o" "o" "o"] ["x" "_" "x"] ["_" "x" "_"]], :ai-marker "o", :player-marker "x"} {:board [["o" "o" "_"] ["x" "o" "x"] ["_" "x" "_"]], :ai-marker "o", :player-marker "x"} {:board [["o" "o" "_"] ["x" "_" "x"] ["o" "x" "_"]], :ai-marker "o", :player-marker "x"} {:board [["o" "o" "_"] ["x" "_" "x"] ["_" "x" "o"]], :ai-marker "o", :player-marker "x"})
```

Comment:
This is a pretty difficult test to understand. Generally I feel like testing two sets of large data structures against each other can be pretty hard to debug, since the smallest difference can be hard to track down sometimes.

Unrelated to the readability of this test, but do you believe this is something that should be tested here? Are external pieces relying on your generation of possible game states, or is that just implementation details to your scoring functions?

http://googletesting.blogspot.com/2013/08/testing-on-toilet-test-behavior-not.html